src/tir/transforms/lower_tvm_builtin.cc

```diff
@@ -98,6 +98,15 @@ class BuiltinLower : public StmtExprMutator {
     }
   }
 
+  Stmt VisitStmt_(const LetStmtNode* op) final {
```

Comment:
Why does mutation happens in `LetStmtNode` instead of `CallNode` directly?

Reply:
In this case I mutate the let
```
let buffer = texture2d_alloca(...)
```
to include stmts to check that the return value of the call is not nullptr, to include the free of the allocation in a sequence, and to check for non-zero return value of the free. This is behaving similar to the handling of the Allocate visitor.

I'm quite open to alternatives if there is a better approach that I missed.